traits/trait_types.py

```diff
@@ -1829,192 +1960,180 @@ def init_fast_validator(self, *args):
-                high = eval(self._high)
-                if (low is None) and (high is None):
-                    if isinstance(value, RangeTypes):
-                        self._set_value(object, name, value)
-                        return
-                else:
-                    new_value = self._typed_value(value, low, high)
-                    if (
-                        (low is None)
-                        or (self._exclude_low and (low < new_value))
-                        or ((not self._exclude_low) and (low <= new_value))
-                    ) and (
-                        (high is None)
-                        or (self._exclude_high and (high > new_value))
-                        or ((not self._exclude_high) and (high >= new_value))
-                    ):
-                        self._set_value(object, name, new_value)
-                        return
-            except:
-                pass
-
-        self.error(object, name, value)
-
-    def _typed_value(self, value, low, high):
-        """ Returns the specified value with the correct type for the current
-            dynamic range.
-        """
-        vtype = self._vtype
-        if vtype is None:
-            if low is not None:
-                vtype = type(low)
-            elif high is not None:
-                vtype = type(high)
-            else:
-                vtype = lambda x: x
-
-        return vtype(value)
+        value = self._validate(object, name, value)
 
-    def _set_value(self, object, name, value):
-        """ Sets the specified value as the value of the dynamic range.
-        """
         cname = "_traits_cache_" + name
         old = object.__dict__.get(cname, Undefined)
         if old is Undefined:
-            old = eval(self._value)
+            # Again, for consistency with the non-clipped case we should really
+            # be validating here. See the comment in _get.
+            old = self._get_default_value(object)
         object.__dict__[cname] = value
         if value != old:
             object.trait_property_changed(name, old, value)
 
     def full_info(self, object, name, value):
         """ Returns a description of the trait.
         """
-        if self._vtype is not Undefined:
-            low = eval(self._low)
-            high = eval(self._high)
-            low, high = (
-                self._typed_value(low, low, high),
-                self._typed_value(high, low, high),
-            )
-        else:
-            low = self._low
-            high = self._high
+        low, high = self._get_bounds(object)
+
+        value_info = self._value_trait.full_info(object, name, value)
 
         if low is None:
             if high is None:
-                return self._type_desc
+                return value_info
 
             return "%s <%s %s" % (
-                self._type_desc,
-                "="[self._exclude_high :],
+                value_info,
+                "="[self._exclude_high:],
```

Comment:
Drive-by fixup: turn this and similar lines into if-expressions like `'' if self._exclude_high else '='`?  Possibly even pre-calculate before building the string at the cost of slightly more work.

Reply:
Actually, `'<' if self._exclude_high else '<=' would be even better (with the appropriate change to the string.

Reply:
I've reworked this, and added tests. It may well be a bit slower than it was, but this is unlikely to be a performance-critical path. I *was* sorely tempted to use f-strings in the rework, but so far this PR has remained Python 2 compatible, and there hasn't been a compelling enough reason to change that.